DBMS/DBMS-Session3/assignment3_sql.txt

```diff
@@ -0,0 +1,106 @@
+SHOW DATABASES;
+USE ShopCart;
+SHOW TABLES;
+
+#Display Shopper’s information along with number of orders he/she placed during last 30 days
+SELECT 
+    u.user_id,
+    u.user_name,
+    u.user_email,
+    (SELECT 
+            COUNT(o.order_id)
+        FROM
+            user_order o
+        WHERE
+            u.user_id = o.user_id
+                AND o.order_date >= DATE(NOW()) - INTERVAL 30 DAY) AS total_orders
+FROM
+    user u
+WHERE
+    u.roles_id = 1;
+    
+
+#Display the top 10 Shoppers who generated maximum number of revenue in last 30 days
+SELECT * FROM user_order;
+SELECT 
+	user_id,
+	order_amount
+FROM 
+	user_order
+WHERE 
+	order_date >= DATE(NOW()) - INTERVAL 30 DAY
+ORDER BY order_amount DESC
+LIMIT 10;
+
+
+#Display top 20 Products which are ordered most in last 60 days along with numbers
+CREATE VIEW products
+AS
+SELECT DISTINCT 
+	prod_id 
+FROM
+	order_line l, user_order o
+WHERE 
+	l.order_id = o.order_id
+    AND
+	o.order_date >= DATE(NOW()) - INTERVAL 60 DAY;
+
+SELECT
+	p2.prod_id, 
+    p2.prod_title,
+    (SELECT SUM(l.prod_quantity) 
+    FROM order_line l 
+    WHERE p1.prod_id = l.prod_id) AS total_orders
+FROM
+	products p1, product p2
+WHERE
+	p1.prod_id = p2.prod_id
+ORDER BY total_orders DESC;
+
+
+#Display Monthly sales revenue of the StoreFront for last 6 months. It should display each month’s sale
+#? ? ?
+#-----
+#? ? ?
+
+
+#Mark the products as Inactive which are not ordered in last 90 days
```

Comment:
update product p
left join order_line l on p.product_id NOT IN (SELECT DISTINCT product_id from order_line WHERE order_date < today-90)
set p.isActive = false;